pingcap/tidb
# executor: do not log slow query for pessimistic write conflict retry

### What problem does this PR solve?

Problem Summary:
Slow Query is logged for each pessimistic select for update retry statement, which is confusing.

### What is changed and how it works?

What's Changed:
Do not use `recordSet`, execute directly.

How it Works:
`recordSet` log slow query on `Close`, remove it to avoid logging slow query.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
Run a conflicting workload for a while and check that the slow-log file size is smaller compares to the master.
